### ISSUE TITLE
Magisk Versions Post 23.0 Cause Bootloop on Fresh Install

### ISSUE THREAD

**Issue by akano12** [topjohnwu/Magisk#5040]
Device: Galaxy J6+ (SM-J610f)
Android version: 9.0
Magisk version name: 5d6d241
Magisk version code: 23015

I was running 23015 without any problems before flashing stock firmware yesterday,
I attempted to root my phone again by flashing 23015 but my phone bootlooped (it was working fine with 23015 on same firmware), same symptoms as in this report https://github.com/topjohnwu/Magisk/issues/4810#issue-1034511424 , i tried flashing 2-3 other versions with no success, however i flashed 23.0 and the phone booted, reflashed 23015 and phone also booted without issues.

Grabbed those files from /sys/fs/pstore because adb was inaccessible

TL;DR 
flashing 23015 =>bootloop
flashing 23.0 then 23015 =>no bootloop
[dmesg.zip](https://github.com/topjohnwu/Magisk/files/7696997/dmesg.zip)

**Comment by yujincheng08** [comment]
> dm-verity corrupted, printing data in hex

**Comment by canyie** [comment]
What is the latest version that works with fresh installing on your device? 23001?

**Comment by akano12** [comment]
> What is the latest version that works with fresh installing on your device? 23001?

Not sure about the code but it's the one released in may, v23.0.

**Comment by canyie** [comment]
Any luck with Magisk v24? Some bugs has been fixed.

**Comment by akano12** [comment]
> Any luck with Magisk v24? Some bugs has been fixed.

Issues discussed in this thread is only present on fresh install and sadly i can't provide/help with any tests for now as i'm extremely busy during the next 2 weeks.

Will gladly provide test results and logs when i'm free...

**Comment by singhh9596** [comment]
Agreed, some privious as well as the latest build released on 2 March also caused me a blootloop.

**Comment by akano12** [comment]
> Agreed, some privious as well as the latest build v24.2 also caused me a blootloop.

on fresh install or after updating?

**Comment by singhh9596** [comment]
> > Agreed, some privious as well as the latest build v24.2 also caused me a blootloop.
> 
> on fresh install or after updating?

While Updating.

**Comment by akano12** [comment]
> > > Agreed, some privious as well as the latest build v24.2 also caused me a blootloop.
> > 
> > 
> > on fresh install or after updating?
> 
> While Updating.

then i suggest you open a new issue as the problem described here is different from yours.

**Comment by canyie** [comment]
@singhh9596 I suggest you try the latest CI build first, the bug may already be fixed. https://github.com/topjohnwu/Magisk/actions/runs/1932106012

**Comment by osm0sis** [comment]
@akano12 try with stable 24.3, and if that doesn't work, latest Canary 24304 please.

**Comment by canyie** [comment]
Any updates with Magisk 24305? Devs may wanna close stale issues...

**Comment by akano12** [comment]
@osm0sis

I know it's been a while but can you reopen? Or should i open new issue?

**Comment by osm0sis** [comment]
I'll reopen if you post some new logs from the latest canary.

**Comment by akano12** [comment]
> I'll reopen if you test and provide some new logs from the latest canary.


Already tested amd flashing failed, previously flashing would finish without errors but phone bootloops, but now flashing fails and phone bootloops too. 


And sure think i can provide logs, but is there anything to pull from twrp other than /sys/fs/pstore considering the phone is bootlooping?

**Comment by akano12** [comment]
@osm0sis (sorry for the pings)

Tried flashing latest canary and flashing failed, also note that flashing 23.0 works well, after phone boots with v23 i can upgrade to lastest canary without issues, initial flashing with anything post 23 fails.

Below you can find dmesg and recovery logs
[dmesg.log](https://github.com/user-attachments/files/15523098/dmesg.log)
[recovery.log](https://github.com/user-attachments/files/15523099/recovery.log)

**Comment by canyie** [comment]
This looks like a new regression compared to the original one, maybe it's better to open a new issue rather than reopening this one 🤔
Btw, I remember I received similar issues before, maybe already fixed on master?

**Comment by osm0sis** [comment]
@akano12 hit Reference in new issue on your last reply

**Comment by akano12** [comment]
> This looks like a new regression compared to the original one, maybe it's better to open a new issue rather than reopening this one 🤔
> Btw, I remember I received similar issues before, maybe already fixed on master?

Does magisk have github actions releases? Or do i have to build myself?

And sure i can open new issue (but will test what you suggested before, maybe it's resolved)

**Comment by Lsudo2024** [comment]
***@***.***
